virttest/qemu_vm.py

```diff
@@ -1038,10 +1038,36 @@ def add_memorys(devices, params):
                 options.append("maxmem=%s" % mem_params["maxmem"])
                 if mem_params.get("slots"):
                     options.append("slots=%s" % mem_params["slots"])
+            sgx_epc_memdev_id = dict()
             for name in params.objects("mem_devs"):
                 dev = devices.memory_define_by_params(params, name)
+                for ele in dev:
+                    if isinstance(ele, qdevices.Memory) and \
+                            ele.params["backend"] == "memory-backend-epc":
+                        sgx_epc_memdev_id[name] = ele.params["id"]
                 devs.extend(dev)
+            sgx_epc_cmd = ""
+            cap_type = "sgx-epc"
+            for idx, ele in enumerate(params.objects("vm_sgx_epc_devs")):
+                sgx_epc_params = params.object_params(ele)
+                sgx_epc_memdev_ele = sgx_epc_params["vm_sgx_epc_memdev"]
+                if sgx_epc_memdev_ele not in sgx_epc_memdev_id.keys():
```

Comment:
sgx_epc_memdev_id.keys() -> sgx_epc_memdev_id, others look good to me

Reply:
Thanks Zhenchao, worked for me, code updated.